### ISSUE TITLE
Type mismatch error not reported on inconvertible expression in object initializer with incomplete member assignment

### ISSUE THREAD

**Issue by Rekkonnect** [dotnet/roslyn#74954]
## Version Used
VS 2022 Version 17.11.2
Roslyn 4.11.0 (5e3a11e2e7)

Project info
C# 12.0
.NET 8.0

## Steps to Reproduce

```csharp
public class C
{
    public Make M()
    {
        return new()
        {
            Name = 123,
            Value =,
        }
    }
}

public class Make
{
    public string Name { get; set; }
    public int Value { get; set; }
}
```

In [SharpLab](https://sharplab.io/#v2:EYLgtghglgdgNAFxBAzguATEBqAPgAQGYACfAJmIGEBYAKAG87jnSSBZCAawFNi2AKAJRMWjWiwmkA7MRjcA7kJGTiYlSoByEMLwC8xAIxlCcZeuYA1CABsArntPj1AXzOva7ukVIUOPOmos3vgGAAzEWjqqxADm3AgA3MQo8UnuEt6wCMRWdrz0sanJRe7uQA==)

In the above example, `123` cannot be assigned to `Name` because `int` cannot be converted to `string`.

## Expected Behavior

An error is reported on `123`.

## Actual Behavior

No error is reported, and instead only a nullability warning may appear (if nullable is enabled). Despite the nullability annotations, the warning is reported.

## Notes

This does not happen if the actual name of the type is provided, like `Make`. So, based on the above example, if we write
```csharp
        return new Make()
        {
            Name = 123,
            Value =,
        }
```
the error would be correctly reported on `123`.

**Comment by CyrusNajmabadi** [comment]
If there are no parse errors, is the semantic error reported?

**Comment by Rekkonnect** [comment]
Yes

**Comment by CyrusNajmabadi** [comment]
Seems fine to me then. You fix the error and find out about the other one.
